app/src/main/java/com/noalecohen/dispatcher/repository/ArticlesRepository.kt

```diff
@@ -1,14 +1,27 @@
 package com.noalecohen.dispatcher.repository
 
 import com.noalecohen.dispatcher.api.news.NewsServiceApiController
-import com.noalecohen.dispatcher.model.response.News
-import retrofit2.Call
+import com.noalecohen.dispatcher.model.response.Article
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
 
 class ArticlesRepository {
     private val apiController = NewsServiceApiController()
 
-    fun fetchTopHeadlinesByCountry(country: String): Call<News> {
-        return apiController.fetchTopHeadlinesByCountry(country)
+    suspend fun fetchTopHeadlinesByCountry(country: String): List<Article>? {
+
+        return withContext(Dispatchers.IO) {
```

Comment:
You should specify the thread when creating the CorutineScope and if you want to move to another thread use the withcontext function.